### PR TITLE
BE: Create CardChecklistPlugin / ChecklistSchema

### PR DESCRIPTION
**Pull request by uche-okoroafor**
BE: Create CardChecklistPlugin /ChecklistSchema  #50 

### What this PR does (required):
 - it can  create, get, update, patch, and delete a checklist using the Cardchecklist plugin and checklist schema   

### Screenshots / Videos (front-end only):
- Post at least one screenshot to show the feature in action. **Only required in front-end work**

### Any information needed to test this feature (required):
- Post what steps to take to be able to test this feature (eg. "sign up as a new user and upload a profile picture from the navbar")

### Any issues with the current functionality (optional):
- List any problems where you were not able to complete all tasks on ticket
- List any problems this PR may cause for the project or other active PRs
- Post a screenshot/video of the problem, along with detailed console outputs where applicable

### CODE REVIEW

**Review by ezeilo-su — COMMENTED** [review]
LGTM!

File: server/models/Checklist.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+ChecklistPluginSchema.methods.get = async ({ params }) => {
+  const { checklistItemId } = params;
+  if (!checklistItemId) {
+    throw new Error("checklistItemId is not defined");
+  }
+  const data = await this.model("ChecklistPlugin").findById(checklistItemId);
+  if (data) {
+    return { status: 200, data };
+  }
+  throw new Error("something went wrong");
```
Comment (moffatethan) [inline server/models/Checklist.js:33]: I would rather have this respond with a new status code or 500. 

`{ response: 400, message }`

Comment (moffatethan) [reply server/models/Checklist.js:33]: Please update where applicable in the action handlers.

**Review by moffatethan — CHANGES_REQUESTED** [review]

File: server/models/Checklist.js [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    };
+  }
+  throw new Error("something went wrong");
+};
+
+ChecklistPluginSchema.methods.update = async ({ body }) => {
+  const { checklistItem, checklistItemId, isChecked } = body;
+  if (!checklistItemId || !isChecked || !checklistItem) {
+    throw new Error("required params is not defined");
+  }
```
Comment (moffatethan) [inline server/models/Checklist.js:59]: We should be responding with an object

`{ response: XXX, message: "" }`

Please address where applicable

**Review by moffatethan — CHANGES_REQUESTED** [review]

File: server/controllers/dist/check.dev.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+var asyncHandler = require("express-async-handler");
+
+var _require = require("../models/Checklist"),
+  ChecklistSchema = _require.ChecklistSchema; // @route POST /auth/register
+// @desc Register user
+// @access Public
+
+exports.check = asyncHandler(function _callee(req, res, next) {
+  var user;
```
Comment (moffatethan) [inline server/controllers/dist/check.dev.js:13]: Please remove the built code from the PR

**Review by moffatethan — CHANGES_REQUESTED** [review]

**Review by moffatethan — APPROVED** [review]

**Review by moffatethan — COMMENTED** [review]
**Please address merge conflicts before merge**

**Pull request merged**